django_project/frontend/src/components/Map/LeftSide/Layers/index.tsx

```diff
@@ -98,10 +98,29 @@ export default function Layers({
       });
       if (!res.ok) throw new Error("Export endpoint returned error");
 
-      const { task_id } = await res.json();
-      setExportTasks((prev) => ({ ...prev, [layer.id]: task_id }));
+      const { task_id, already_exported, cog_id, download_url } = await res.json();
+      const exportNrtLayer: ExportNrtLayer = {
+        loading: false,
+        error: null,
+        layerId: layer.id,
+        taskId: task_id,
+        status: already_exported ? "completed" : "processing",
+        download_url: download_url,
+        cogId: cog_id,
+      };
+
+      dispatch(setExportNrtLayer(exportNrtLayer));
+
+      if (already_exported) {
+        showToast(
+          "Export already finished",
```

Comment:
@danangmassandy I think we should start wrapping the static text to support localization, but we can do that later.